backend/canvas_app_explorer/management/commands/manage_pylti.py

```diff
@@ -0,0 +1,66 @@
+        parser.add_argument('--client_id', dest='client_id', type=int, required=True, help="Canvas LTI Client ID")
+        parser.add_argument('--title', dest='title', required=True, type=str, help="LTI Title")
+        parser.add_argument('--tool_key', dest='tool_key', required=True, type=str,
+                            help="Name of Tool Key to use, will create if new")
+        parser.add_argument('--deployment_ids', dest='deployment_ids',
+                            nargs='*', type=str, help="List of Deployment ID(s). Can be multiple.", default="")
+
+    def handle(self, *args, **options: dict):
+        key: RsaKey = RSA.generate(4096)
+
+        self.stdout.write('Generating public and private key for LTI Tool Key table')
+
+        platform = options["platform"]
+
+        # If platform doesn't contain https add it
+        if not platform.startswith("https://"):
+            platform = f"https://{platform}"
+
+        auth_login_url = f"{platform}/api/lti/authorize_redirect"
+        auth_token_url = f"{platform}/login/oauth2/token"
+        issuer = platform
+        key_set_url = f"{platform}/api/lti/security/jwks"
+
+        # This will try to create the tool key and if it already exists it will look it up instead
+        try:
+            lti_key = LtiToolKey.objects.create(name=options["tool_key"],
+                                                private_key=key.exportKey().decode('utf-8'),
+                                                public_key=key.publickey().exportKey().decode('utf-8')
+                                                )
+        except IntegrityError:
+            # Key must already have been created
+            self.stdout.write(
+                f"Using existing key named {options['tool_key']}")
+            # Lookup the tool key
+            lti_key = LtiToolKey.objects.get(name=options["tool_key"])
+
+        # Update or create a value based on the client_id and issuer keys
+        LtiTool.objects.update_or_create(client_id=options["client_id"],
+                                         issuer=issuer,
+                                         defaults=dict(title=options["title"],
+                                                       is_active=True,
+                                                       client_id=options["client_id"],
+                                                       issuer=issuer,
+                                                       use_by_default=False,
+                                                       auth_login_url=auth_login_url,
+                                                       auth_token_url=auth_token_url,
+                                                       key_set_url=key_set_url,
+                                                       tool_key=lti_key,
+                                                       deployment_ids=options["deployment_ids"]
+                                                       ))
```

Comment:
I was thinking of having some of this semi self contained so I could contribute it back to the pylti project. But this is really Canvas specific and we'd need different patterns for other LMSes. This was really just intended as an easier way than running SQL queries or editing the DB with an IDE to get this initial row in here that didn't currently exist. I was trying to figure this out for MyLA too so we could move the configuration there to the DB, but MyLA not as likely to be multi-tenant as this app will hopefully someday be. 

The only semi-annoying part of doing it without the script is generating the RSA keys. MyLA does have a utility for that but I usually just use the IMS website. I was just hoping this would save some time for local dev.

Reply:
I think this could be done later if we write a test for this, but I'd like to try to get it back upstream. Just going to resolve for now. I refactored this since this change too.